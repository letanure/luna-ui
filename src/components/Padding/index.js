import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Padding component.
 * E.g.:
 * ```html
 * <Padding selector="elements" value="1px 3px 4px 2px" />
 * ```
 */
class PaddingComponent extends React.PureComponent {
  render() {
    const { selector, value } = this.props;

    const Padding = styled.div`
      & ${selector} {
        padding: ${value};
      }
    `;

    return <Padding {...this.props}>{this.props.children}</Padding>;
  }
}

PaddingComponent.propTypes = {
  children: PropTypes.node,
  /**
   * Selects the element to receive padding value
   */
  selector: PropTypes.string,
  /**
   * Value of padding to selected elements
   */
  value: PropTypes.string
};

PaddingComponent.defaultProps = {
  children: null,
  selector: '*',
  value: '11px'
};

export default PaddingComponent;
