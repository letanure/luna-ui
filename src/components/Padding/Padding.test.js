import React from 'react';
import renderer from 'react-test-renderer';

import Padding from './index.js';

describe('<Padding />', () => {
  const content =
    <h1>Snap Test Rocks</h1>
  ;

  const component =
    <Padding>{content}</Padding>;

  const componentSelector =
    <Padding selector="h1">{content}</Padding>;

  const componentValue =
    <Padding value="10px 20px 20px 30px">{content}</Padding>;

  const componentSelectAndValue =
    <Padding selector="h1" value="10px 20px 20px 30px">{content}</Padding>;

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Selector', () => {
    const tree = renderer
      .create(componentSelector)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Value', () => {
    const tree = renderer
      .create(componentValue)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Selector and Value', () => {
    const tree = renderer
      .create(componentSelectAndValue)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
