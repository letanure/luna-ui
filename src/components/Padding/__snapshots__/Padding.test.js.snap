// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Padding /> toMatchSnapshot 1`] = `
<div
  className="sc-bdVaJa crQnAS"
  value="11px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;

exports[`<Padding /> toMatchSnapshot: Selector 1`] = `
<div
  className="sc-bwzfXH icXqRl"
  value="11px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;

exports[`<Padding /> toMatchSnapshot: Selector and Value 1`] = `
<div
  className="sc-bxivhb iosSTl"
  value="10px 20px 20px 30px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;

exports[`<Padding /> toMatchSnapshot: Value 1`] = `
<div
  className="sc-htpNat dpiLti"
  value="10px 20px 20px 30px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;
